apps/wpcom-block-editor/src/wpcom/features/redirect-onboarding-user-after-publishing-post.js

```diff
@@ -1,43 +1,48 @@
 import { dispatch, select, subscribe } from '@wordpress/data';
-import domReady from '@wordpress/dom-ready';
 import { getQueryArg } from '@wordpress/url';
+import { useEffect } from 'react';
+import useSiteIntent from './use-site-intent';
 
 const START_WRITING_FLOW = 'start-writing';
 
-export function redirectOnboardingUserAfterPublishingPost() {
-	const isStartWritingFlow = getQueryArg( window.location.search, START_WRITING_FLOW ) === 'true';
+export function RedirectOnboardingUserAfterPublishingPost() {
+	const { siteIntent: intent } = useSiteIntent();
 
-	if ( ! isStartWritingFlow ) {
+	useEffect( () => {
+		if ( intent === START_WRITING_FLOW ) {
+			dispatch( 'core/edit-post' ).closeGeneralSidebar();
+			document.documentElement.classList.add( 'start-writing-hide' );
+		}
+	}, [ intent ] );
```

Comment:
I put this in useEffect so it only runs once when the page loads. That way the user can open and close the sidebars if they choose. Otherwise the user can click on the sidebar buttons, but they're closed immediately after the click.

Reply:
Thanks for improving that!